src/parser.rs

```diff
@@ -290,6 +290,46 @@ impl<'a> Parser<'a> {
         }
     }
 
+    // Parse an array index as a natural number
+    fn array_index(&mut self) -> Option<String> {
```

Comment:
I think there's a method to parse integers somewhere else, perhaps that could be leveraged here?